distributed/scheduler.py

```diff
@@ -3422,9 +3422,20 @@ def heartbeat_worker(
         ws._last_seen = time()
 
         if executing is not None:
-            ws._executing = {
-                parent._tasks[key]: duration for key, duration in executing.items()
-            }
+            ws._executing = {}
+            for key, duration in executing.items():
```

Comment:
IIUC this is a `double`. I think Cython may be able to figure out the rest once this is typed.

```suggestion
            duration: double
            for key, duration in executing.items():
```